ledger/participant-integration-api/src/main/resources/db/migration/postgres-appendonly/V115__activate_string_interning_for_template_ids.sql

```diff
@@ -0,0 +1,221 @@
+--  Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+--  SPDX-License-Identifier: Apache-2.0
+
+SELECT 'Template Id interning data migration - Preparation...';
+
+-- add temporary index to aid string lookups as migrating data
+CREATE UNIQUE INDEX string_interning_external_string_temp_idx ON string_interning USING btree (external_string);
+
+-- drop participant_events view (enables column type changes)
+DROP VIEW participant_events;
+
+-- add temporary migration function
+CREATE FUNCTION internalize_string(prefix TEXT, string TEXT)
+RETURNS INTEGER
+LANGUAGE plpgsql
+AS
+$$
+DECLARE
+   result INTEGER;
+BEGIN
+   EXECUTE
+   'SELECT CASE' ||
+   '  WHEN $2 IS NULL THEN NULL' ||
+   '  ELSE (' ||
+   '    SELECT string_interning.internal_id' ||
+   '      FROM string_interning' ||
+   '      WHERE $1 || $2 = string_interning.external_string' ||
+   '  )' ||
+   'END'
+   INTO result
+   USING prefix, string;
+   ASSERT
+     (string IS NULL AND result IS NULL) OR (string IS NOT NULL AND result IS NOT NULL),
+     'Could not internalize the string: ' || string;
+   RETURN result;
+END
+$$;
+
+
+SELECT 'Template Id interning data migration - Migrating participant_events_divulgence...';
+
+DROP INDEX participant_events_divulgence_template_id_idx;
+
+ALTER TABLE participant_events_divulgence
+  ALTER COLUMN template_id TYPE INTEGER
+  USING internalize_string('t|', template_id);
+
+
+
+SELECT 'Template Id interning data migration - Migrating participant_events_create...';
+
+DROP INDEX participant_events_create_template_id_idx;
+
+ALTER TABLE participant_events_create
+  ALTER COLUMN template_id TYPE INTEGER
+  USING internalize_string('t|', template_id);
```

Comment:
Did you measure the performance of this?

Reply:
Not yet, it is planned to measure this together with the party interning and the filter table conversion together